coredns/datadog_checks/coredns/coredns.py

```diff
@@ -0,0 +1,108 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from datadog_checks.checks.prometheus import PrometheusCheck
+from datadog_checks.errors import CheckException
+
+
+class CoreDNSCheck(PrometheusCheck):
+    """
+    Collect CoreDNS metrics from Prometheus
+    """
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(CoreDNSCheck, self).__init__(name, init_config, agentConfig, instances)
+        self.NAMESPACE = 'coredns'
+
+        self.metrics_mapper = {
+            # Primarily, metrics are emitted by the prometheus plugin: https://coredns.io/plugins/metrics/
+            # Note: the count metrics were moved to specific functions
+            # below to be submitted as both gauges and monotonic_counts
+            'coredns_dns_request_duration_seconds': 'request_duration.seconds',
+            'coredns_dns_request_size_bytes': 'request_size.bytes',
+            # https://coredns.io/plugins/proxy/
+            'coredns_proxy_request_duration_seconds': 'proxy_request_duration.seconds',
+            # https://coredns.io/plugins/cache/
+            'coredns_cache_size': 'cache_size.count',
+        }
+        self.ignore_metrics = {
```

Comment:
do we wanna ignore all of these? (people can find lots of metrics useful for a lot of reasons)